src/serializers/infer.rs

```diff
@@ -91,6 +92,16 @@ pub(crate) fn infer_to_python_known(
         Ok::<PyObject, PyErr>(new_dict.into_py(py))
     };
 
+    let serialize_with_serializer = |value: &PyAny, is_model: bool| {
```

Comment:
I was sort of just trying to follow the pattern above with `serialize_dict`, not sure if this is the best way to do it, open to alternative suggestions. (@samuelcolvin feel free to make changes directly on this branch if you prefer.)

Also, in principle, we could use something like this as a fallback for any not-otherwise-serializable object that has a `__pydantic_serializer__` attribute of type `SchemaSerializer`. I didn't do that here as I didn't have any good use case in mind and it seemed unnecessary anyway since the current logic will still use this path as long as a `__pydantic_validator__` is also present.